packages/netglade_utils/lib/src/extensions/iterable_extensions.dart

```diff
@@ -0,0 +1,20 @@
+import 'dart:core';
+
+import 'package:collection/collection.dart';
+
+typedef Selector<T, K> = K Function(T selectFrom);
+
+extension IterableExtensions<T> on Iterable<T> {
+  /// Returns a set containing only distinct elements from the given collection.
+  Set<T> distinct() {
+    return toSet();
+  }
+
+  /// Returns a list containing only elements from the given collection having distinct keys returned by the given [selector] function. If multiple elements have the same key, first element is returned.
```

Comment:
Nevrací to list, ale iterable. 

Pro lepší čitelnost se to snažíme zalamovat dle https://sembr.org